ins-node/src/wifi_module/wifi_module.ino

```diff
@@ -1,10 +1,156 @@
 #include <stdint.h>
+#include <math.h>
+#include <vector>
+#include <utility>
+#include <ESP8266WiFi.h>
+#include "wifi_credentials.h"
 
-const uint8_t WIFI_LED = 4;
+ESP8266WiFiClass wifi;
+EspClass espPower;
+const uint8_t DEVICE_ID = 1; // The identification for the specific node
+const int8_t CONNECTION_RETRIES = 20; // The maximum amount of wifi connection attempts
+const uint8_t DEFAULT_TRANSMISSION_SIZE = 10;
+uint8_t currentTransmissionSize = DEFAULT_TRANSMISSION_SIZE;
+
+const uint8_t TX_PIN = 10;
+const uint8_t RX_PIN = 9;
+
+/**
+   Gets the network context (i.e. MAC address and RSSI value) of the specified
+   network item.
+   @param networkItem The network item index of which to fetch the context
+   @return            A tuple of (MAC address, RSSI)
+*/
+std::pair <String, int32_t> getNetworkContext(int8_t networkItem) {
```

Comment:
What String class is used here? std::? Should be included in header file to remove dependency on external files.

Reply:
No, this is not the std::string, but the [Arduino String class](https://www.arduino.cc/reference/en/language/variables/data-types/stringobject/). Every Arduino sketch "secretly" includes an `#include <Arduino.h>` on top of it, which is added by the IDE preprocessor.